exercises/concept/take-a-number-deluxe/.meta/exemplar.ex

```diff
@@ -0,0 +1,82 @@
+defmodule TakeANumberDeluxe do
+  use GenServer
+
+  # Client API
+
+  def start_link(init_arg) do
+    GenServer.start_link(__MODULE__, init_arg)
+  end
+
+  def report_state(machine) do
+    GenServer.call(machine, :report_state)
+  end
+
+  def queue_new_number(machine) do
+    GenServer.call(machine, :queue_new_number)
+  end
+
+  def serve_next_queued_number(machine, priority_number \\ nil) do
+    GenServer.call(machine, {:serve_next_queued_number, priority_number})
+  end
+
+  def reset_state(machine) do
+    GenServer.cast(machine, :reset_state)
+  end
+
+  # Server callbacks
+
+  @impl true
```

Comment:
I know it's most common to use `@impl true`, but I've also read that using `@impl GenServer` might be better, as it's more explicit. Not sure what's best.
I guess it should mainly be addressed in the behaviour concept.